### PR TITLE
Documentation improvements on the executor

### PR DESCRIPTION
**Pull request by mjcarroll**
Add documentation to a few undocumented methods.

### CODE REVIEW

File: rclcpp/include/rclcpp/executor.hpp [hunk 1]
```diff
@@ -315,6 +315,16 @@ class Executor
   virtual void
   spin_all(std::chrono::nanoseconds max_duration);
 
+
+  /// Collect work once and execute the next available work, optionally within a duration.
+  /**
+   * This function can be overridden.  The dfault implementation is suitable for
```
Comment (clalancette) [inline rclcpp/include/rclcpp/executor.hpp:321]:
```suggestion
   * This function can be overridden.  The default implementation is suitable for
```

File: rclcpp/include/rclcpp/executor.hpp [hunk 2]
```diff
@@ -413,12 +423,29 @@ class Executor
   is_spinning();
 
 protected:
+  /// Add a node to executor, execute the next available unit of work, and remove the node.
+  /**
+   * Implementation of spin_node_once using std::chrono::nanoseconds
+   * \param[in] node Shared pointer to the node to add.
+   * \param[in] timeout How long to wait for work to become available. Negative values cause
+   * spin_node_once to block indefinitely (the default behavior). A timeout of 0 causes this
+   * function to be non-blocking.
```
Comment (clalancette) [inline rclcpp/include/rclcpp/executor.hpp:432]: Probably indent these lines a little further?  At least, that is what is done above.

File: rclcpp/include/rclcpp/executor.hpp [hunk 3]
```diff
@@ -502,16 +564,54 @@ class Executor
...
   bool
   get_next_ready_executable(AnyExecutable & any_executable);
 
+  /// Check for executable in ready state and populate union structure.
+  /**
+   * This is the implementation of get_next_ready_executable that takes into
+   * account the current state of callback groups' association with nodes and
+   * executors.
+   *
+   * This checks in a particular order for avabilable work:
```
Comment (clalancette) [inline rclcpp/include/rclcpp/executor.hpp:583]:
```suggestion
   * This checks in a particular order for available work:
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI.

**Review by fujitatomoya — APPROVED** [review]

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18218/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12749/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18887/)

**Pull request merged**